diff --git a/node_modules/@react-native-firebase/remote-config/ios/RNFBConfig/RNFBConfigModule.m b/node_modules/@react-native-firebase/remote-config/ios/RNFBConfig/RNFBConfigModule.m
index bbef694..edd6f41 100644
--- a/node_modules/@react-native-firebase/remote-config/ios/RNFBConfig/RNFBConfigModule.m
+++ b/node_modules/@react-native-firebase/remote-config/ios/RNFBConfig/RNFBConfigModule.m
@@ -15,6 +15,7 @@
  *
  */
 
+#import <React/RCTBridgeModule.h>
 #import <Firebase/Firebase.h>
 #import <React/RCTConvert.h>
 #import <React/RCTUtils.h>
